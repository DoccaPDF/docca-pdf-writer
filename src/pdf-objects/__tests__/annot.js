import Annot from '../annot';
import Action from '../action';

const action = Action({
  S: 'URI', URI: 'http://docca.io'
});

const annot = Annot({
  id: 42,
  A: action,
  Subtype: 'Link',
  Rect: [10, 20, 30, 40],
  H: 'I'
});

describe('pdf-objects', () => {
  describe('annot', () => {
    it('creates a new object', () => {
      expect(annot).toMatchSnapshot();
    });

    it('flattens keys values', () => {
      expect(annot.flattenKeys()).toMatchSnapshot();
    });

    it('returns a PDF object', () => {
      expect(annot.toPDF()).toMatchSnapshot();
    });
  });
});
