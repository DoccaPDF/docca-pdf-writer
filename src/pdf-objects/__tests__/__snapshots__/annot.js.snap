// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pdf-objects annot creates a new object 1`] = `
Object {
  "A": Object {
    "S": "URI",
    "URI": "http://docca.io",
  },
  "H": "I",
  "Rect": Array [
    10,
    20,
    30,
    40,
  ],
  "Subtype": "Link",
  "id": 42,
}
`;

exports[`pdf-objects annot flattens keys values 1`] = `
Object {
  "A": "<<
/S /URI
/URI (http://docca.io)
>>
",
  "BS": undefined,
  "C": undefined,
  "Dest": undefined,
  "F": undefined,
  "H": "/I",
  "Rect": "[10 20 30 40]",
  "Subtype": "/Link",
  "Type": "/Annot",
}
`;

exports[`pdf-objects annot returns a PDF object 1`] = `
"42 0 obj
<<
/Type /Annot
/A <<
/S /URI
/URI (http://docca.io)
>>

/Subtype /Link
/Rect [10 20 30 40]
/H /I
>>
endobj
"
`;
